### PR TITLE
EWPP-3273: Update Carousel pattern and paragraph rendering.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 1]
```diff
@@ -945,20 +945,31 @@ public function testCarousel(): void {
     $this->container->get('router.builder')->rebuild();
 
     // Create English files.
-    $en_file_1 = file_save_data(file_get_contents(\Drupal::service('extension.list.theme')->getPath('oe_theme') . '/tests/fixtures/example_1.jpeg'), 'public://example_1_en.jpeg');
+    $en_file_1 = \Drupal::service('file.repository')->writeData(file_get_contents(\Drupal::service('extension.list.theme')->getPath('oe_theme') . '/tests/fixtures/example_1.jpeg'), 'public://example_1_en.jpeg');
```
Comment (nagyad) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]: $this->container->

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 2]
```diff
@@ -945,20 +945,31 @@ public function testCarousel(): void {
...
+
+    $en_file_2 = \Drupal::service('file.repository')->writeData(file_get_contents(\Drupal::service('extension.list.theme')->getPath('oe_theme') . '/tests/fixtures/example_1.jpeg'), 'public://example_2_en.jpeg');
     $en_file_2->setPermanent();
     $en_file_2->save();
+    // Get second english file styled URI.
+    $en_file_2_uri = $style->buildUri($en_file_2->getFileUri());
 
     // Create Bulgarian files.
-    $bg_file_1 = file_save_data(file_get_contents(\Drupal::service('extension.list.theme')->getPath('oe_theme') . '/tests/fixtures/example_1.jpeg'), 'public://example_1_bg.jpeg');
+    $bg_file_1 = \Drupal::service('file.repository')->writeData(file_get_contents(\Drupal::service('extension.list.theme')->getPath('oe_theme') . '/tests/fixtures/example_1.jpeg'), 'public://example_1_bg.jpeg');
```
Comment (nagyad) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]: $this->container->

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 3]
```diff
@@ -945,20 +945,31 @@ public function testCarousel(): void {
...
     $bg_file_1->setPermanent();
     $bg_file_1->save();
-    $bg_file_2 = file_save_data(file_get_contents(\Drupal::service('extension.list.theme')->getPath('oe_theme') . '/tests/fixtures/example_1.jpeg'), 'public://example_2_bg.jpeg');
+    // Get first bulgarian file styled URI.
+    $bg_file_1_uri = $style->buildUri($bg_file_1->getFileUri());
+
+    $bg_file_2 = \Drupal::service('file.repository')->writeData(file_get_contents(\Drupal::service('extension.list.theme')->getPath('oe_theme') . '/tests/fixtures/example_1.jpeg'), 'public://example_2_bg.jpeg');
```
Comment (nagyad) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]: $this->container->

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 4]
```diff
@@ -1023,39 +1034,44 @@ public function testCarousel(): void {
...
+    // Assert the carousel size.
+    $carousel_size = $crawler->filter('section.ecl-banner--l');
+    $this->assertCount(4, $carousel_size);
     $assert = new CarouselAssert();
     $expected_values = [
       'items' => [
         [
           'title' => 'Item 1',
-          'image' => file_create_url($en_file_1->getFileUri()),
+          'image' => \Drupal::service('file_url_generator')->generateAbsoluteString($en_file_1_uri),
```
Comment (nagyad) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]: $this->container->

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 5]
```diff
@@ -1023,39 +1034,44 @@ public function testCarousel(): void {
...
           'variant' => 'text-highlight',
         ],
         [
           'title' => 'Item 2',
           'description' => 'Item description 2',
           'url' => 'http://www.example.com/',
           'url_text' => 'CTA 2',
-          'image' => file_create_url($en_file_2->getFileUri()),
+          'image' => \Drupal::service('file_url_generator')->generateAbsoluteString($en_file_2_uri),
```
Comment (nagyad) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]: $this->container->

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 6]
```diff
@@ -1023,39 +1034,44 @@ public function testCarousel(): void {
...
           'variant' => 'text-highlight',
         ],
         [
           'title' => 'Item 3',
-          'image' => file_create_url($en_file_1->getFileUri()),
+          'image' => \Drupal::service('file_url_generator')->generateAbsoluteString($en_file_1_uri),
```
Comment (nagyad) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]: $this->container->

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 7]
```diff
@@ -1023,39 +1034,44 @@ public function testCarousel(): void {
...
           'variant' => 'text-highlight',
         ],
         [
           'title' => 'Item 4',
           'description' => 'Item description 4',
           'url' => '/',
           'url_text' => 'CTA 4',
-          'image' => file_create_url($en_file_2->getFileUri()),
+          'image' => \Drupal::service('file_url_generator')->generateAbsoluteString($en_file_2_uri),
```
Comment (nagyad) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]: $this->container->

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 8]
```diff
@@ -1064,19 +1080,23 @@ public function testCarousel(): void {
 
     // Assert paragraph rendering for Bulgarian version.
     $html = $this->renderParagraph($paragraph, 'bg');
+    $crawler = new Crawler($html);
+    // Assert the carousel size.
+    $carousel_size = $crawler->filter('section.ecl-banner--l');
+    $this->assertCount(4, $carousel_size);
     $expected_values['items'][0]['title'] = 'BG Item 1';
-    $expected_values['items'][0]['image'] = file_create_url($bg_file_1->getFileUri());
+    $expected_values['items'][0]['image'] = \Drupal::service('file_url_generator')->generateAbsoluteString($bg_file_1_uri);
```
Comment (nagyad) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]: $this->container->

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 9]
```diff
@@ -1064,19 +1080,23 @@ public function testCarousel(): void {
...
     $expected_values['items'][1]['title'] = 'BG Item 2';
     $expected_values['items'][1]['description'] = 'BG Item description 2';
     $expected_values['items'][1]['url_text'] = 'BG CTA 2';
-    $expected_values['items'][1]['image'] = file_create_url($bg_file_2->getFileUri());
+    $expected_values['items'][1]['image'] = \Drupal::service('file_url_generator')->generateAbsoluteString($bg_file_2_uri);
```
Comment (nagyad) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]: $this->container->

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 10]
```diff
@@ -1064,19 +1080,23 @@ public function testCarousel(): void {
...
     $expected_values['items'][2]['title'] = 'BG Item 3';
-    $expected_values['items'][2]['image'] = file_create_url($bg_file_1->getFileUri());
+    $expected_values['items'][2]['image'] = \Drupal::service('file_url_generator')->generateAbsoluteString($bg_file_1_uri);
```
Comment (nagyad) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]: $this->container->

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 11]
```diff
@@ -1064,19 +1080,23 @@ public function testCarousel(): void {
...
     $expected_values['items'][3]['title'] = 'BG Item 4';
     $expected_values['items'][3]['description'] = 'BG Item description 4';
     $expected_values['items'][3]['url'] = 'http://www.example.com/';
     $expected_values['items'][3]['url_text'] = 'BG CTA 4';
-    $expected_values['items'][3]['image'] = file_create_url($bg_file_2->getFileUri());
+    $expected_values['items'][3]['image'] = \Drupal::service('file_url_generator')->generateAbsoluteString($bg_file_2_uri);
```
Comment (nagyad) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]: $this->container->

**Review by nagyad — CHANGES_REQUESTED** [review]

**Review by nagyad — DISMISSED** [review]

File: oe_theme.theme [hunk 12]
```diff
@@ -2063,21 +2063,10 @@ function oe_theme_preprocess_paragraph__oe_carousel(array &$variables): void {
...
-      $resource_ref = $media->get($field_name)->value;
-      $uri = 'avportal://' . $resource_ref . '.jpg';
-    }
-    // The uri might be empty if the source is of type Image and the file entity
-    // was deleted.
-    if (empty($uri)) {
-      $cacheability->applyTo($variables);
-      continue;
-    }
+    $thumbnail = $media->get('thumbnail')->first();
```
Comment (imanoleguskiza) [inline oe_theme.theme]: Why are we using the thumbnail all of a sudden? We can just create the styled image by loading it and using the $uri we already generated no?
    $style = \Drupal::entityTypeManager()->getStorage('image_style')->load('oe_theme_full_width');
   $style->buildUrl($uri)

**Review by imanoleguskiza — CHANGES_REQUESTED** [review]

**Review by imanoleguskiza — DISMISSED** [review]

**Review by nagyad — APPROVED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Pull request merged**